### DISCUSSION TITLE
Why does Istio have issues with GRPC + headless services?

### DISCUSSION THREAD

**Discussion by mrehm96** [istio/istio#52972] [Q&A]
For setups where Istio needs to work with GRPC + headless services it is only possible to make it work by telling Istio that it should work with TCP instead of GRPC directly (e.g., see https://grafana.com/docs/loki/latest/setup/install/istio/). Am I right with this assumption and if I am, what is the reason why it is like that?

**Comment by howardjohn** [comment]
This is plausibly to change from host header to IP based matching? See https://istio.io/latest/docs/ops/configuration/traffic-management/traffic-routing/. This is a loki specific thing though, its not universal that gRPC needs changes like this.

**Reply by mrehm96** [reply]
Would it be a plausible solution to just set the authority header manually for any GRPC call (using withAuthority; see https://pkg.go.dev/google.golang.org/grpc) making problems in this case or does this introduce undesired behavior?

**Reply by howardjohn** [reply]
Certainly. The original link was about `loki` though which may not allow that customization.

FWIW, in my experience typically applications already set the authority (without needing withAuthority) to a value that Istio understands. Its plausible loki is just doing something somewhat abnormal
